### ISSUE TITLE
Consider LTS versions when upgrading Node.js docker image

### ISSUE THREAD

**Issue by fgreinacher** [renovatebot/renovate#13270]
### What would you like Renovate to be able to do?

Renovate should consider LTS semantics when upgrading Node.js Docker images.

See https://github.com/renovatebot/renovate/discussions/10163 for more context.

### If you have any ideas on how this should be implemented, please tell us here.

We could add something like this to `config:base`

```json
{
      "matchDatasources": ["docker"],
      "matchPackageNames": ["node"],
      "versioning": "node"
}
```

This would not work out-of-the-box when people use images like `node:16-alpine`, so we also "need to update node versioning to treat that suffix as compatibility. I've never seen node prereleases so that behavior should be safe." (quoting @viceice).

### Is this a feature you are interested in implementing yourself?

Maybe

**Comment by rarkins** [comment]
Very closely related: https://github.com/renovatebot/renovate/issues/5990

My idea is to separate the concept of compatibility from version, so that e.g. instead of docker versioning being compatibility-aware, instead we split typical docker tags into version-compatibility.

**Comment by ErisDS** [comment]
We've had a bunch of weird PRs from renovate bumping Node to v17 when we have our configuration set like this:

```
"node": {
  "supportPolicy": [
    "lts"
  ]
},
```

Here's an example:

https://github.com/TryGhost/gscan/pull/408/files

Is this bug related to this issue or should I raise a separate issue?

**Comment by rarkins** [comment]
`supportPolicy` is a deprecated/removed option, but I don't think it would ever have prevented alpine-flavored major updates either. This is not a bug, but instead a missing feature, waiting on #5990

**Comment by Rochet2** [comment]
In case someone is looking for a workaround, one can be found [here](https://docs.renovatebot.com/configuration-options/#versioncompatibility). Hiding in plain sight.

> ```
>   "packageRules": [
>     {
>       "matchDatasources": ["docker"],
>       "matchPackageNames": ["node"],
>       "versionCompatibility": "^(?<version>[^-]+)(?<compatibility>-.*)?$",
>       "versioning": "node"
>     }
>   ]
> ```

**Comment by rarkins** [comment]
I think we're nearly there, the main question before closing this is: should Renovate have such a rule by default, or perhaps as part of the workarounds preset, which is included by most with config:recommended?

**Comment by fgreinacher** [comment]
> should Renovate have such a rule by default, or perhaps as part of the workarounds preset,

I think it would work well as part of the workarounds preset.

**Comment by kubmir** [comment]
II tried to use @Rochet2 's workaround and it works but it updates my node from 18 to 21 not to 20 which is LTS. I also tried solution with regex manager described [here](https://github.com/renovatebot/renovate/discussions/14909#discussioncomment-2642053) and it still updates to current version of node not LTS. My dockerfile is named like `dockerfile` and is in the root of project.

Any ideas why ?

```{
   "extends":[
      "config:base",
      ":enablePreCommit",
      ":semanticCommits"
   ],
   "regexManagers":[
      {
         "fileMatch":[
            "^dockerfile$"
         ],
         "matchStrings":[
            "FROM --platform=linux/amd64 node:(?<currentValue>.*?)-alpine"
         ],
         "depNameTemplate":"node",
         "datasourceTemplate":"node",
         "versioningTemplate":"node"
      }
   ],
   "packageRules":[
      {
         "groupName":"non-major updates",
         "matchUpdateTypes":[
            "patch",
            "minor"
         ]
      },
      {
         "matchPackageNames":[
            "node"
         ],
         "matchManagers":[
            "dockerfile"
         ],
         "enabled":false
      }
   ]
}
